apps/condo/domains/common/hooks/useDownloadFileFromServer.ts

```diff
@@ -0,0 +1,65 @@
+import { useCallback } from 'react'
+import { notification } from 'antd'
+
+import { useIntl } from '@open-condo/next/intl'
+
+
+type FileType = {
+    url: string
+    name: string
+}
+type DownloadFileType = (file: FileType) => Promise<void>
+type UseDownloadFileFromServerType = () => { downloadFile: DownloadFileType }
+
+const ERROR_DOWNLOAD_FILE = 'Failed to download file'
+const DIRECT_DOWNLOAD_FILE_TYPES_REGEX = /.*\.(svg|html|htm|xml|txt)$/i
+
+export const useDownloadFileFromServer: UseDownloadFileFromServerType = () => {
```

Comment:
Do we have cases when no redirect is presented in response?
For example, with different file adapter…

Reply:
Yes, there are indeed such cases. (For example local)
I think this should be fixed in a separate issue